app/src/main/java/com/x8bit/bitwarden/data/platform/repository/SettingsRepositoryImpl.kt

```diff
@@ -337,6 +338,56 @@ class SettingsRepositoryImpl(
                     ?: DEFAULT_IS_SCREEN_CAPTURE_ALLOWED,
             )
 
+    override val allSettingsBadgeCountFlow: StateFlow<Int>
+        get() = combine(
+            allSecuritySettingsBadgeCountFlow,
+            allAutofillSettingsBadgeCountFlow,
+            transform = ::sumSettingsBadgeCount,
+        )
+            .stateIn(
+                scope = unconfinedScope,
+                started = SharingStarted.Lazily,
+                initialValue = 0,
+            )
+
+    override val allSecuritySettingsBadgeCountFlow: StateFlow<Int>
+        get() = activeUserId
+            ?.let {
+                // can be expanded to support multiple security settings
+                getShowUnlockBadgeFlow(userId = it)
```

Comment:
Would it make more sense to use the `authDisjkSource.activeUserIdChangesFlow` and then `flatMapLatest`

Reply:
yeah for sure, I was following the pattern for `vaultLastSyncStateFlow`, should I update at as well? Or that out of scope ya think?

Reply:
Hmmm, We probably should. maybe a followup PR.